src/ui/public/markdown/markdown.js

```diff
@@ -56,70 +60,46 @@ export function markdownFactory(whiteListedRules, openLinksInNewTab = false) {
       return originalLinkRender(tokens, idx, options, env, self);
     };
   }
-
-  return markdownIt;
-}
-
-export class Markdown extends Component {
-  constructor(props) {
-    super(props);
-
-    this.markdownIt = markdownFactory(this.props.whiteListedRules, this.props.openLinksInNewTab);
-
-    this.state = {
-      renderedMarkdown: this.transformMarkdown(this.props),
-    };
-  }
-
   /**
-   * This method is used to actually render markdown from the passed parameter
+   * This method is used to render markdown from the passed parameter
    * into HTML. It will just return an empty string when the markdown is empty.
-   * Since we want to use this with dangerouslySetInnerHTML, this method returns
-   * the required object format with an __html key in it.
+   * @param {String} markdown - Rhe markdown String
+   * @return {Object} - Returns the dangerouslySetInnerHTML required object
+   * with the rendered HTML stored in the __html key.
    */
-  transformMarkdown(params) {
-    if (!params.markdown) {
+  return (markdown) => {
+    if (!markdown) {
       return { __html: '' };
     }
-    return { __html: this.markdownIt.render(params.markdown) };
-  }
-
-  componentWillReceiveProps(props) {
-    const hasOpenLinksInNewTabChanged = props.openLinksInNewTab !== this.props.openLinksInNewTab;
-    const hasMarkdownChanged = props.markdown !== this.props.markdown;
-    const hasWhiteListerRulesChanged = props.whiteListedRules !== this.props.whiteListedRules;
-
-    if (hasOpenLinksInNewTabChanged || hasWhiteListerRulesChanged) {
-      this.markdownIt = markdownFactory(props.whiteListedRules, props.openLinksInNewTab);
-    }
-    if (hasMarkdownChanged || hasOpenLinksInNewTabChanged || hasWhiteListerRulesChanged) {
-      this.setState({
-        renderedMarkdown: this.transformMarkdown(props),
-      });
-    }
-  }
+    return { __html: markdownIt.render(markdown) };
+  };
+}, (whiteListedRules = [], openLinksInNewTab = false) => {
+  return whiteListedRules.join('_').concat(openLinksInNewTab);
+});
 
+export class Markdown extends PureComponent {
```

Comment:
I would suggest, we just use a pure function component instead inheriting `PureComponent` here. Functional of course no difference, but I think it reads a bit cleaner and aligns better with what we usually do in components.

Reply:
I'm using a PureComponent to take advantage of its shallow compare of props. This avoids calling the render method if the parent rerenders and the props aren't changed as opposed to what happens when using a pure function that renders every time its parent renders independently from the props and state.
We have the `markdownit` render method in the `render` method of the pure component so it's better to don't rerender everything every time.